accounts/backends.py

```diff
@@ -35,16 +37,23 @@ def authenticate(self, request, remote_user, shibboleth_attributes):
         User = get_user_model()
         username = self.clean_username(remote_user)
         user, created = User.objects.get_or_create(username=username)
+
+        # Add inital attributes on first log in
         if created:
             user.set_unusable_password()
             if isinstance(shibboleth_attributes, Mapping):
                 for key, value in shibboleth_attributes.items():
-                    if value:
-                        setattr(user, key, value)
-                    else:
-                        setattr(user, key, self.searchPennDirectory(username, key))
-
+                    if key != 'affiliation':
+                        setattr(user, key, value if value else self.searchPennDirectory(username, key))
             user.save()
             user = self.configure_user(request, user)
 
+        # Update affiliations with every log in
+        if shibboleth_attributes is not None and 'affiliation' in shibboleth_attributes:
```

Comment:
Probably could move the not-`None` check to encompass the `if created` branch as well. Not sure exactly how shibboleth_attributes is populated, but would it be a successful login if it *was* `None`?

Reply:
See comment below